src/components/shared/TreeView.js

```diff
@@ -45,30 +49,34 @@ const TreeViewHeader = ({ fixedColumns, mainColumn }: TreeViewHeaderProps) =>
 
 function reactStringWithHighlightedSubstrings(
   string: string,
-  substring: string | null,
+  substrings: ?(string[]),
   className: string
 ) {
-  if (!substring) {
+  if (!substrings || !substrings.length) {
     return string;
   }
-  const lowercaseString = string.toLowerCase();
-  const result = [];
-  let startAt = 0;
-  let nextOccurrence = -1;
-  while (
-    (nextOccurrence = lowercaseString.indexOf(substring, startAt)) !== -1
-  ) {
-    const afterNextOccurrence = nextOccurrence + substring.length;
-    result.push(string.substring(startAt, nextOccurrence));
-    result.push(
-      <span key={nextOccurrence} className={className}>
-        {string.substring(nextOccurrence, afterNextOccurrence)}
+
+  const regexpStr = substrings.map(escapeStringRegexp).join('|');
+  const re = new RegExp(regexpStr, 'gi');
+
+  const highlighted = [];
+  let lastOccurrence = 0;
+  let result;
+  while ((result = re.exec(string))) {
+    const typedResult: RegExpResult = result;
+    if (typedResult === null) {
+      break;
+    }
+    highlighted.push(string.substring(lastOccurrence, typedResult.index));
+    lastOccurrence = re.lastIndex;
+    highlighted.push(
+      <span key={typedResult.index} className={className}>
+        {typedResult[0]}
       </span>
     );
-    startAt = afterNextOccurrence;
   }
-  result.push(string.substring(startAt));
-  return result;
+  highlighted.push(string.substring(lastOccurrence));
+  return highlighted;
```

Comment:
The general logic didn't change, only how we find the occurrences: with a regexp.